ast/src/parsed/mod.rs

```diff
@@ -613,25 +613,38 @@ pub trait SourceReference {
 }
 
 macro_rules! impl_source_reference {
+    // Version for types with generic parameter
+    ($enum:ident<$generic:ident>, $($variant:ident),*) => {
+        impl<$generic> SourceReference for $enum<$generic> {
+            impl_source_reference_inner!($enum, $($variant),*);
+        }
+    };
+    // Version for types without generic parameter
     ($enum:ident, $($variant:ident),*) => {
```

Comment:
Maybe not in this PR, but I was not aware that the first item in the list is special. Maybe we should change the syntax to
`impl_source_reference!(PilStatement { Include, Namespace, ... });` (i.e. but the variantes inside `{}`)

Reply:
Another idea could be to think about a derive macro...

Reply:
Yeah, I think this is quite a lot of work.